### ISSUE TITLE
Implement unary_union

### ISSUE THREAD

**Issue by stuartlynn** [kylebarron/geopolars#20]
This should take a GeoSeries, and return the union of all of the geometries within that series.

Is this implemented in the Geo crate: No but some discussion here https://github.com/georust/geo/issues/80 . We might also want to look at the [clipping crate](https://docs.rs/clipping/0.1.1/clipping/gh/struct.CPolygon.html) mention in that thread
GeoPandas equivalent: [unary_union](https://geopandas.org/en/stable/docs/reference/api/geopandas.GeoSeries.unary_union.html#geopandas.GeoSeries.unary_union)

**Comment by kylebarron** [comment]
> We might also want to look at the [clipping crate](https://docs.rs/clipping/0.1.1/clipping/gh/struct.CPolygon.html) mention in that thread

Note that that's GPL3, which I'd like to avoid

**Comment by kylebarron** [comment]
I think it might be implemented in this PR: https://github.com/georust/geo/pull/835

**Comment by stuartlynn** [comment]
Ah good to know about the clipping crate. 

Looks like the georust version is being merged tomorrow which is exciting!

**Comment by stuartlynn** [comment]
This just got merged to the main branch on Geo so we should be able to add it next release

**Comment by kylebarron** [comment]
Looks like there will be a release soon: https://github.com/georust/geo/pull/840. But also seems fine for us to use a git revision of `geo` for development

**Comment by stuartlynn** [comment]
Sounds good. Will write a quick implementation of this with a git tag and we can change it up for the release when it happens.

**Comment by frewsxcv** [comment]
hello there. geo 0.21.0 is out

**Comment by stuartlynn** [comment]
Amazing! Thanks for letting us know !

**Comment by kylebarron** [comment]
Relevant:
https://discord.com/channels/598002550221963289/598002550221963291/985637742367080548

> If anyone is looking for a pretty tricky issue to bang their head against, feel free to have a go at implementing unary union using the new union operation and the approach sketched out here: https://github.com/georust/rstar/issues/80#issuecomment-988615807

**Comment by kylebarron** [comment]
> I was able to implement a unary_union for polygons [JosiahParry/geo-rtree-issue@`main`/src/erm.rs](https://github.com/JosiahParry/geo-rtree-issue/blob/main/src/erm.rs?rgh-link-date=2023-01-29T02%3A26%3A47Z)

I don't think this is the same as a geopandas unary union. If you look at [this SO answer](https://stackoverflow.com/a/40386377/7319250) for example, it merges several input polygons into **one** output polygon. In contrast, your code merges all the input polygons into one output **MultiPolygon** which hasn't been merged together yet
